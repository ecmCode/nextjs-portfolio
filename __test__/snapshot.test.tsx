import { render } from '@testing-library/react';
import Home from '../src/pages/index';
import React from 'react';
import NotFound from '@/pages/404';
import Blogs from '@/pages/blogs';
import About from '@/pages/about';
import Projects from '@/pages/projects';
 
it('renders Home page unchanged', () => {
  const { container } = render(<Home posts={[]} />);
  expect(container).toMatchSnapshot();
});

it('renders 404 page unchanged', () => {
  const { container } = render(<NotFound/>);
  expect(container).toMatchSnapshot();
});

it('renders About page unchanged', () => {
  const { container } = render(<About />);
  expect(container).toMatchSnapshot();
});

it('renders Projects page unchanged', () => {
  const { container } = render(<Projects />);
  expect(container).toMatchSnapshot();
});

it('renders Blogs page unchanged', () => {
  const { container } = render(<Blogs posts={[]} />);
  expect(container).toMatchSnapshot();
});