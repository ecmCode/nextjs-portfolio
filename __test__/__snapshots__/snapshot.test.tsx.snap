// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 404 page unchanged 1`] = `
<div>
  <main>
    <h1>
      404
    </h1>
    <p>
      Page not found
    </p>
    <a
      href="/"
    >
      Home
    </a>
    <br />
  </main>
</div>
`;

exports[`snapshot renders About page unchanged 1`] = `
<div>
  <main>
    <h1>
      About Me
    </h1>
    <div
      class="flex flex-col gap-4"
    >
      <div>
        <h4
          class="flex gap-2 items-center"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 3a1 1 0 011-1h2.153a1 1 0 01.986.836l.74 4.435a1 1 0 01-.54 1.06l-1.548.773a11.037 11.037 0 006.105 6.105l.774-1.548a1 1 0 011.059-.54l4.435.74a1 1 0 01.836.986V17a1 1 0 01-1 1h-2C7.82 18 2 12.18 2 5V3z"
            />
          </svg>
          012345678
        </h4>
        <h4
          class="flex gap-2 items-center"
        >
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          example@example.com
        </h4>
      </div>
      <div
        class="flex gap-10 text-3xl"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0C5.829 0 5.556.01 4.703.048 3.85.088 3.269.222 2.76.42a3.917 3.917 0 0 0-1.417.923A3.927 3.927 0 0 0 .42 2.76C.222 3.268.087 3.85.048 4.7.01 5.555 0 5.827 0 8.001c0 2.172.01 2.444.048 3.297.04.852.174 1.433.372 1.942.205.526.478.972.923 1.417.444.445.89.719 1.416.923.51.198 1.09.333 1.942.372C5.555 15.99 5.827 16 8 16s2.444-.01 3.298-.048c.851-.04 1.434-.174 1.943-.372a3.916 3.916 0 0 0 1.416-.923c.445-.445.718-.891.923-1.417.197-.509.332-1.09.372-1.942C15.99 10.445 16 10.173 16 8s-.01-2.445-.048-3.299c-.04-.851-.175-1.433-.372-1.941a3.926 3.926 0 0 0-.923-1.417A3.911 3.911 0 0 0 13.24.42c-.51-.198-1.092-.333-1.943-.372C10.443.01 10.172 0 7.998 0h.003zm-.717 1.442h.718c2.136 0 2.389.007 3.232.046.78.035 1.204.166 1.486.275.373.145.64.319.92.599.28.28.453.546.598.92.11.281.24.705.275 1.485.039.843.047 1.096.047 3.231s-.008 2.389-.047 3.232c-.035.78-.166 1.203-.275 1.485a2.47 2.47 0 0 1-.599.919c-.28.28-.546.453-.92.598-.28.11-.704.24-1.485.276-.843.038-1.096.047-3.232.047s-2.39-.009-3.233-.047c-.78-.036-1.203-.166-1.485-.276a2.478 2.478 0 0 1-.92-.598 2.48 2.48 0 0 1-.6-.92c-.109-.281-.24-.705-.275-1.485-.038-.843-.046-1.096-.046-3.233 0-2.136.008-2.388.046-3.231.036-.78.166-1.204.276-1.486.145-.373.319-.64.599-.92.28-.28.546-.453.92-.598.282-.11.705-.24 1.485-.276.738-.034 1.024-.044 2.515-.045v.002zm4.988 1.328a.96.96 0 1 0 0 1.92.96.96 0 0 0 0-1.92zm-4.27 1.122a4.109 4.109 0 1 0 0 8.217 4.109 4.109 0 0 0 0-8.217zm0 1.441a2.667 2.667 0 1 1 0 5.334 2.667 2.667 0 0 1 0-5.334z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.545 2.907a13.227 13.227 0 0 0-3.257-1.011.05.05 0 0 0-.052.025c-.141.25-.297.577-.406.833a12.19 12.19 0 0 0-3.658 0 8.258 8.258 0 0 0-.412-.833.051.051 0 0 0-.052-.025c-1.125.194-2.22.534-3.257 1.011a.041.041 0 0 0-.021.018C.356 6.024-.213 9.047.066 12.032c.001.014.01.028.021.037a13.276 13.276 0 0 0 3.995 2.02.05.05 0 0 0 .056-.019c.308-.42.582-.863.818-1.329a.05.05 0 0 0-.01-.059.051.051 0 0 0-.018-.011 8.875 8.875 0 0 1-1.248-.595.05.05 0 0 1-.02-.066.051.051 0 0 1 .015-.019c.084-.063.168-.129.248-.195a.05.05 0 0 1 .051-.007c2.619 1.196 5.454 1.196 8.041 0a.052.052 0 0 1 .053.007c.08.066.164.132.248.195a.051.051 0 0 1-.004.085 8.254 8.254 0 0 1-1.249.594.05.05 0 0 0-.03.03.052.052 0 0 0 .003.041c.24.465.515.909.817 1.329a.05.05 0 0 0 .056.019 13.235 13.235 0 0 0 4.001-2.02.049.049 0 0 0 .021-.037c.334-3.451-.559-6.449-2.366-9.106a.034.034 0 0 0-.02-.019Zm-8.198 7.307c-.789 0-1.438-.724-1.438-1.612 0-.889.637-1.613 1.438-1.613.807 0 1.45.73 1.438 1.613 0 .888-.637 1.612-1.438 1.612Zm5.316 0c-.788 0-1.438-.724-1.438-1.612 0-.889.637-1.613 1.438-1.613.807 0 1.451.73 1.438 1.613 0 .888-.631 1.612-1.438 1.612Z"
          />
        </svg>
      </div>
    </div>
    <div
      class="w-full md:w-4/5 lg:w-2/5 flex flex-col gap-10 p-10"
      data-testid="paragraphs"
    >
      <p>
        Hi there! My name is Lee and I'm a self-taught web developer traveling around the world while working remotely. 
      </p>
      <p>
        I started my programming journey in 2019. Throughout the years I've developed a passion for building websites that are both functional and beautiful. I've always been interested in technology and how it can be used to create amazing things.
      </p>
      <p>
        I specialize in HTML, CSS, JavaScript, Typescript, and React. I have been always looking for new ways to improve my skills and stay up-to-date with the latest trends in web development.
      </p>
      <p>
        My goal as a web developer is to create websites that are easy to use and visually appealing.I believe that a great website should not only look good but also be intuitive and user-friendly.
      </p>
      <p>
        That's why I pay attention to every detail of the design and functionality of each website I build. I'm a firm believer in the power of collaboration, and I enjoy working closely with my clients to make sure their vision comes to life.
      </p>
      <p>
        When I'm not working on websites, you can usually find me exploring the city or trying out new restaurants. I'm also an avid reader and enjoy learning about different cultures and languages.
      </p>
      <p>
        If you're interested in working together or just want to say hi, feel free to reach out to me through my contact page!
      </p>
    </div>
    <a
      href="/"
    >
      <button
        class="px-4 py-2 bg-slate-200 dark:bg-slate-900"
      >
        Home
      </button>
    </a>
  </main>
</div>
`;

exports[`snapshot renders Blogs page unchanged 1`] = `
<div>
  <main>
    <h1>
      Blogs
    </h1>
    <ul
      class="grid grid-cols-1 lg:grid-cols-2 w-full sm:w-2/3 gap-10 my-10"
    />
  </main>
</div>
`;

exports[`snapshot renders Home page unchanged 1`] = `
<div>
  <main>
    <div
      class="w-auto z-0 mb-10 lg:mb-0"
    >
      <h1
        class="title"
      >
        Portfolio
      </h1>
      <div
        class="wrapper"
      >
        <img
          alt="A fantasy portal icon"
          class="icon"
          data-nimg="1"
          decoding="async"
          draggable="false"
          fetchpriority="high"
          height="400"
          src="/_next/image?url=%2Fimg.jpg&w=828&q=100"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=100 1x, /_next/image?url=%2Fimg.jpg&w=828&q=100 2x"
          style="color: transparent;"
          width="400"
        />
        <div
          class="content"
        >
          <h2>
            A collection of my work.
          </h2>
          <p
            class="paragraph"
          >
            
                I'm Lee, a self taught frontend developer since 2019. I specialize in responsive and user-friendly websites. Whether you're a small business or a large corporation, I'm here to help bring your vision to life. Feel free to explore my portfolio and contact me to discuss your project.
                
          </p>
          <div
            class="flex gap-10"
            data-testid="buttons"
          >
            <a
              href="/projects"
            >
              <button
                class="btn"
              >
                Projects
              </button>
            </a>
            <a
              href="#contact"
            >
              <button
                class="btn"
              >
                Contact
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full py-6 bg-gradient-to-br from-slate-50/10 to-slate-300/30 "
    >
      <h2
        class="px-10"
      >
        About Me
      </h2>
    </div>
    <div
      class="w-full"
    >
      <div
        class="min-h-[33vh]"
      >
        <div
          class="p-4"
        >
          <div
            class="px-10 md:px-40 py-10 flex flex-col gap-6"
          >
            <p>
              Hi there! My name is Lee and I'm a self-taught web developer traveling around the world while working remotely. 
            </p>
            <p>
              I started my programming journey in 2019. Throughout the years I've developed a passion for building websites that are both functional and beautiful. I've always been interested in technology and how it can be used to create amazing things.
            </p>
            <p>
              I specialize in HTML, CSS, JavaScript, Typescript, and React. I have been always looking for new ways to improve my skills and stay up-to-date with the latest trends in web development.
            </p>
            <div
              class="text-right"
            >
              <a
                class="bg-slate-500/20 rounded-sm py-2 px-6"
                href="/about"
              >
                Read More
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full py-6 bg-gradient-to-br from-slate-50/10 to-slate-300/30 "
    >
      <h2
        class="px-10"
      >
        Techs
      </h2>
    </div>
    <div
      class="w-full py-2 px-10 bg-gradient-to-b from-slate-700/20 to-slate-800/20"
    >
      <h4
        class="px-10 uppercase tracking-widest"
      >
        Stacks
      </h4>
    </div>
    <div
      class="w-full"
    >
      <div
        class="min-h-[33vh]"
      >
        <div
          class="p-4"
        >
          <p>
            HTML
          </p>
          <p>
            CSS
          </p>
          <p>
            Javascript
          </p>
          <p>
            Typescript
          </p>
          <p>
            React
          </p>
        </div>
      </div>
    </div>
    <div
      class="w-full py-2 px-10 bg-gradient-to-b from-slate-700/20 to-slate-800/20"
    >
      <h4
        class="px-10 uppercase tracking-widest"
      >
        Projects
      </h4>
    </div>
    <div
      class="w-full"
    >
      <div
        class="min-h-[33vh]"
      >
        <div
          class="p-4"
        >
          <ul
            class="w-full grid grid-cols-1 md:grid-cols-3 gap-4"
          >
            <li
              class="flex justify-between gap-4 bg-cyan-400/20 p-2 uppercase tracking-wider select-none"
            >
              countdown clock
              <a
                href="https://ecmcode.github.io/countdown-timer/"
                target="_blank"
              >
                <button
                  class="text-cyan-400 font-semibold underline"
                >
                  Demo
                </button>
              </a>
            </li>
            <li
              class="flex justify-between gap-4 bg-cyan-400/20 p-2 uppercase tracking-wider select-none"
            >
              shopping list
              <a
                href="https://ecmcode.github.io/shopping-list/"
                target="_blank"
              >
                <button
                  class="text-cyan-400 font-semibold underline"
                >
                  Demo
                </button>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="w-full py-6 bg-gradient-to-br from-slate-50/10 to-slate-300/30 "
    >
      <h2
        class="px-10"
      >
        Blogs
      </h2>
    </div>
    <div
      class="w-full"
    >
      <div
        class="min-h-[33vh]"
      >
        <div
          class="p-4"
        >
          <div
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4"
          />
          <div
            class="text-center mt-10"
          >
            <a
              class="bg-slate-500/20 rounded-sm py-2 px-6"
              href="/blogs"
            >
              All Blogs
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full py-6 bg-gradient-to-br from-slate-50/10 to-slate-300/30 "
    >
      <h2
        class="px-10"
      >
        Contact Me
      </h2>
    </div>
    <div
      id="contact"
    >
      Contact Me
    </div>
  </main>
</div>
`;

exports[`snapshot renders Projects page unchanged 1`] = `
<div>
  <main>
    <h1>
      Projects
    </h1>
    <ul
      class="grid grid-cols-1 md:grid-cols-2 justify-center items-center place-items-center w-full h-auto gap-4 py-10"
    >
      <li
        class="flex flex-col gap-4 justify-between items-center w-2/3 h-full p-10 bg-slate-500/50 rounded-lg"
      >
        <h2>
          countdown clock
        </h2>
        <p>
          countdown clock app using React + Typescript
        </p>
        <div
          class="flex justify-around w-full"
        >
          <a
            href="https://ecmcode.github.io/countdown-timer/"
            target="_blank"
          >
            <button>
              Demo
            </button>
          </a>
          <a
            href="https://github.com/ecmCode/countdown-timer"
            target="_blank"
          >
            <button>
              Code
            </button>
          </a>
        </div>
      </li>
      <li
        class="flex flex-col gap-4 justify-between items-center w-2/3 h-full p-10 bg-slate-500/50 rounded-lg"
      >
        <h2>
          shopping list
        </h2>
        <p>
          shopping list app using React
        </p>
        <div
          class="flex justify-around w-full"
        >
          <a
            href="https://ecmcode.github.io/shopping-list/"
            target="_blank"
          >
            <button>
              Demo
            </button>
          </a>
          <a
            href="https://github.com/ecmCode/shopping-list"
            target="_blank"
          >
            <button>
              Code
            </button>
          </a>
        </div>
      </li>
    </ul>
  </main>
</div>
`;
